### PR TITLE
feature/mx-1618 mex drop UI for endnote

### PR DESCRIPTION
**Pull request by ZehraVictoria**
# PR Context
- Start the App with` pdm run drop run` before running the tests locally.
- please check if tests are sufficient (tested mostly directly in the UI)

# Added
- reflex mini-UI in mex-drop for library team to send endnote XMLs

### CODE REVIEW

File: tests/test_main.py [hunk 1]
```diff
@@ -13,3 +19,78 @@ def test_index(page: Page) -> None:
...
+            "mex.drop.state.DropSettings.get",
+            return_value=MagicMock(drop_directory="/mock/path"),
+        ),
+    ):
+        await app_state.submit_data(form_data=form_data)
+
+        mock_write_to_file.assert_called_once_with(
+            b"content1", pathlib.Path("/mock/path/system1/file1.xml")
+        )
+        assert len(app_state.temp_files) == 0
```
Comment (cutoffthetop) [inline tests/test_main.py:96]: these are actually solid unit tests 👍 but, they are not testing code in `main.py`, so i'd move them to `tests/test_state.py`

and then, could you add a couple of integration tests as well? where we actually upload a file and check it being written to disk. should be quite doable using playwright -> https://playwright.dev/docs/input#upload-files

the test `test_index` already uses playwright for an integration test, but you can also take a peak over to the editor, eg here https://github.com/robert-koch-institut/mex-editor/blob/0.5.0/tests/search/test_main.py

**Review by cutoffthetop — CHANGES_REQUESTED** [review]
very nice and clean! i tried it out locally, and it's working smoothly.
i have just some smaller suggestions and ideas, and a request for some integration tests

File: mex/drop/main.py [hunk 2]
```diff
@@ -17,21 +33,154 @@ def mex_drop_logo() -> rx.Component:
...
+    """Return the index for the drop app.
+
+    Returns:
+        rx.Component: A component with style, event trigger and other props
+    """
+    return rx.box(
+        rx.center(
+            rx.container(
+                rx.hstack(
+                    rx.card(
```
[the remark this answers is not in the document]
Comment (ZehraVictoria) [reply mex/drop/main.py:87]: fixed in 1d4001c5942613367f6f9f5484aa6cc5a79da2b6

File: mex/drop/main.py [hunk 3]
```diff
@@ -17,21 +33,154 @@ def mex_drop_logo() -> rx.Component:
...
+                                    align="center",
+                                ),
+                                id="upload_one",
+                                max_files=100,
+                                accept=ACCEPTED_FILE_FORMATS,
+                                border="1px dotted rgb(107,99,246)",
+                                padding="3em",
+                                align="center",
+                                on_drop=AppState.handle_upload(
+                                    rx.upload_files(upload_id="upload_one")
```
[the remark this answers is not in the document]
Comment (ZehraVictoria) [reply mex/drop/main.py]: fixed in fce9a529f43c4290cb7980ce3d413dd2085d401c

File: mex/drop/state.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+    temp_files: list[TempFile] = []
+    form_data: dict[str, str] = {}
+
+    @classmethod
+    def get_instance(cls) -> Any:
+        """Get the instance of the class."""
+        if cls._instance is None:
+            cls._instance = cls()
+        return cls._instance
```
[the remark this answers is not in the document]
Comment (ZehraVictoria) [reply mex/drop/state.py]: superfluous helper function deleted in fce9a529f43c4290cb7980ce3d413dd2085d401c

**Review by cutoffthetop — APPROVED** [review]
looks good! ready to merge 💯

**Pull request merged**